web/src/hooks/useQueryParams.ts

```diff
@@ -0,0 +1,54 @@
+import { useCallback, useMemo } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+
+// minimal implementation of https://www.npmjs.com/package/use-query-params
+export function useQueryParams<T extends Record<string, string | undefined>>(
```

Comment:
could also use https://www.npmjs.com/package/use-query-params if we're fine with a new external dependency

Reply:
is fine for me to use an external library

Reply:
Ok, I switched to use-query-params. It's actually already included anyway, because it's a dependency of Perses.